### ISSUE TITLE
Incorrect E1136: unsubscriptable-object with pandas DataFrame after it has been previously accessed

### ISSUE THREAD

**Issue by juliangilbey** [PyCQA/pylint#6942]
### Bug description

Here's a short piece of code that illustrates the bug:
```python
import pandas as pd  # type: ignore

df = pd.read_csv("data.csv")

df_a = df.drop(columns=["a"])
df_b = df["b"]
```
pylint gives the error message:
```
pylint_unsubscriptable.py:6:7: E1136: Value 'df' is unsubscriptable (unsubscriptable-object)
```
Removing the `df.drop` line also loses this error message, as does swapping the order of the lines.

There is a comment in #3637 which suggests that the two issues may be closely related.

### Command used

```shell
`pylint -d C0114 pylint_unsubscriptable.py`
```

### Pylint output

```shell
************* Module pylint_unsubscriptable
pylint_unsubscriptable.py:6:7: E1136: Value 'df' is unsubscriptable (unsubscriptable-object)

--------------------------------------------------------------------
Your code has been rated at -2.50/10 (previous run: -5.00/10, +2.50)
```
```


### Expected behavior

Not to give this error message.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.5 (main, Jun  8 2022, 09:26:22) [GCC 11.3.0]
```

### OS / Environment

Debian testing (bookworm/sid)

### Additional dependencies

pandas==1.3.5

**Comment by DanielNoord** [comment]
I think this is a duplicate of https://github.com/PyCQA/pylint/issues/3637 right?

**Comment by juliangilbey** [comment]
It might be, it might not be; I'm not sure what the root cause of #3637 is so I didn't (and still don't) know whether a fix for that would fix my case as well.  Maybe I should have just added this to that issue, in which case I apologise.
